pulp_rpm/app/viewsets.py

```diff
@@ -123,6 +123,124 @@ class RpmRepositoryViewSet(RepositoryViewSet, ModifyRepositoryActionMixin):
     endpoint_name = "rpm"
     queryset = RpmRepository.objects.exclude(user_hidden=True)
     serializer_class = RpmRepositorySerializer
+    queryset_filtering_required_permission = "rpm.view_rpmrepository"
+
+    DEFAULT_ACCESS_POLICY = {
+        "statements": [
+            {
+                "action": ["list"],
+                "principal": ["authenticated"],
+                "effect": "allow",
+                "condition": "has_model_perms:rpm.view_rpmrepository",
+            },
+            {
+                "action": ["retrieve"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:rpm.view_rpmrepository",
+            },
+            {
+                "action": ["create"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_perms:rpm.add_rpmrepository",
+            },
+            {
+                "action": ["update", "partial_update"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:rpm.change_rpmrepository",
+            },
+            {
+                "action": ["modify"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:rpm.change_rpmrepository",
+            },
+            {
+                "action": ["destroy"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:rpm.delete_rpmrepository",
+            },
+            {
+                "action": ["sync"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:rpm.sync_rpmrepository",
+            },
+            {
+                "action": ["list_roles", "add_role", "remove_role"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": ["has_model_or_obj_perms:rpm.manage_roles_rpmrepository"],
+            },
+        ],
+        "creation_hooks": [
+            {
+                "function": "add_roles_for_object_creator",
+                "parameters": {"roles": "rpm.rpmrepository_owner"},
+            }
+        ],
+    }
+
+    # Here are also defined plugin-wide `LOCKED_ROLES`
+    LOCKED_ROLES = {
+        "rpm.rpmrepository_owner": [
+            "rpm.add_rpmrepository",
+            "rpm.change_rpmrepository",
+            "rpm.delete_rpmrepository",
+            "rpm.manage_roles_rpmrepository",
+            "rpm.sync_rpmrepository",
+            "rpm.view_rpmrepository",
+        ],
+        "rpm.rpmrepository_creator": [
+            "rpm.add_rpmrepository",
+            "rpm.change_rpmrepository",
+            "rpm.sync_rpmrepository",
+            "rpm.view_rpmrepository",
```

Comment:
You should not include these. The creator role is by nature a global one (before creating, there is not object to associate with). After you created a repository however you will receive the object permissions to do all these things, via the creation hooks.